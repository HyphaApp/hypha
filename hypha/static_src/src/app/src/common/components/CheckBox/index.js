import * as React from "react";
import PropTypes from 'prop-types';

import HelperComponent from "@common/components/HelperComponent";


const CheckBox = props => {
  return <div className="form__group  form__group--checkbox form__group--wrap">

    <label 
      htmlFor={props.name}
      className="form__question form__question--boolean_field checkbox_input">
        <span>{props.label}</span>
         {props.required ? <span className="form__required"> *</span> : ""}
      </label>


    <div className="form__item" 
      onClick={e => props.onChange(props.name, !props.value)}
    >
      <input 
        type="checkbox" 
        name={props.name}
        id={props.name}
        checked={props.value ? "checked" : ""}
        onChange={e => props.onChange(props.name, e.currentTarget.value)}
      />
      <label />
    </div>
    <hr />
    <HelperComponent {...props.helperProps} />

  </div>
}

CheckBox.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  required: PropTypes.bool,
  onChange: PropTypes.func,
  value: PropTypes.node,
  helperProps: PropTypes.node
}

CheckBox.displayName = 'CheckBox';
export default CheckBox;
