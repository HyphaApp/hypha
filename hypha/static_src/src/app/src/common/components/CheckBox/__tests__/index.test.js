import React from "react";
import { mount, shallow } from "enzyme";
import sinon from "sinon";
import CheckBox from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });


describe("Test CheckBox component", () => {
  const name = "test name";
  const label = "test label";
  const required = false;
  const helperProps = {};
  const value = "1";
  const onChange = jest.fn()

  const subject = mount(<CheckBox
    name={name}
    label={label}
    required={required}
    value={value}
    helperProps={helperProps}
    onChange = {onChange}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have searched classes and span element with passed text", () => {
    expect(subject.find('.form__group').length).toBe(1);
    expect(subject.find('.form__item').length).toBe(1);
    expect(subject.find('form__required').length).toBe(0);
    expect(subject.find('span').text()).toBe(label);
    expect(subject.find('input').prop('checked')).toBe("checked");
    expect(subject.children().length).toEqual(1)
    subject.find('.form__item').props().onClick();
    expect(onChange).toHaveBeenCalledWith(name, false)
    subject.find('input').last().props().onChange({currentTarget : { value : 1}});
    expect(onChange).toHaveBeenCalledWith(name, 1)
    expect(onChange).toHaveBeenCalledTimes(2);
    expect(subject.find('input').props().checked).toEqual("checked")
  });

  test('render a CheckBox component', () => {
    expect(subject).toMatchSnapshot();
  });
});

describe("Test CheckBox component with required", () => {
  const name = "test name";
  const label = "test label";
  const required = true;
  const helperProps = {};
  const value = "1";
  const onChange = jest.fn()

  const subject = mount(<CheckBox
    name={name}
    label={label}
    required={required}
    value={value}
    helperProps={helperProps}
    onChange = {onChange}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have searched classes and element", () => {
    expect(subject.find('.form__required').length).toBe(1);
  });

  test('render a CheckBox component', () => {
    expect(subject).toMatchSnapshot();
  });
});
