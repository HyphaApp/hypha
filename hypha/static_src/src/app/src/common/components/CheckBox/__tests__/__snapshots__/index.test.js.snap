// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CheckBox component render a CheckBox component 1`] = `
<CheckBox
  helperProps={Object {}}
  label="test label"
  name="test name"
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          "test name",
          false,
        ],
        Array [
          "test name",
          1,
        ],
      ],
    }
  }
  required={false}
  value="1"
>
  <div
    className="form__group  form__group--checkbox form__group--wrap"
  >
    <label
      className="form__question form__question--boolean_field checkbox_input"
      htmlFor="test name"
    >
      <span>
        test label
      </span>
    </label>
    <div
      className="form__item"
      onClick={[Function]}
    >
      <input
        checked="checked"
        id="test name"
        name="test name"
        onChange={[Function]}
        type="checkbox"
      />
      <label />
    </div>
    <hr />
    <HelperComponent />
  </div>
</CheckBox>
`;

exports[`Test CheckBox component with required render a CheckBox component 1`] = `
<CheckBox
  helperProps={Object {}}
  label="test label"
  name="test name"
  onChange={[MockFunction]}
  required={true}
  value="1"
>
  <div
    className="form__group  form__group--checkbox form__group--wrap"
  >
    <label
      className="form__question form__question--boolean_field checkbox_input"
      htmlFor="test name"
    >
      <span>
        test label
      </span>
      <span
        className="form__required"
      >
         *
      </span>
    </label>
    <div
      className="form__item"
      onClick={[Function]}
    >
      <input
        checked="checked"
        id="test name"
        name="test name"
        onChange={[Function]}
        type="checkbox"
      />
      <label />
    </div>
    <hr />
    <HelperComponent />
  </div>
</CheckBox>
`;
